go/arrow/util/protobuf_reflect_test.go

```diff
@@ -0,0 +1,247 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"github.com/apache/arrow/go/v17/arrow/array"
+	"github.com/apache/arrow/go/v17/arrow/memory"
+	"github.com/apache/arrow/go/v17/arrow/util/util_message"
+	"github.com/huandu/xstrings"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/anypb"
+	"strings"
+	"testing"
+)
+
+func SetupTest() util_message.AllTheTypes {
+	msg := util_message.ExampleMessage{
+		Field1: "Example",
+	}
+
+	anyMsg, _ := anypb.New(&msg)
+
+	return util_message.AllTheTypes{
+		String_:  "Hello",
+		Int32:    10,
+		Int64:    100,
+		Sint32:   -10,
+		Sin64:    -100,
+		Uint32:   10,
+		Uint64:   100,
+		Fixed32:  10,
+		Fixed64:  1000,
+		Sfixed32: 10,
+		Bool:     false,
+		Bytes:    []byte("Hello, world!"),
+		Double:   1.1,
+		Enum:     util_message.AllTheTypes_OPTION_0,
+		Message:  &msg,
+		Oneof:    &util_message.AllTheTypes_Oneofstring{Oneofstring: "World"},
+		Any:      anyMsg,
+		//Breaks the test as the `RecordFromJson` randomises the order
+		//SimpleMap:   map[int32]string{99: "Hello", 100: "World", 98: "How", 101: "Are", 1: "You"},
```

Comment:
RecordFromJSON doesn't randomize the order, Golang maps have a non-deterministic order.